solvers/Electrostatic2D3V/CMakeLists.txt

```diff
@@ -30,8 +30,9 @@ target_include_directories(
 
 target_link_libraries(
   ${ELECTROSTATIC_2D3V_LIBRARY_NAME}
-  PRIVATE ${NEKTAR++_LIBRARIES} ${NEKTAR++_TP_LIBRARIES} ${MPI_LIBRARIES}
-          ${NESO_PARTICLES_LIBRARIES} ${NESO_PARTICLES_LINK_FLAGS})
+  PRIVATE ${NESO_LIBRARY_NAME} ${NEKTAR++_LIBRARIES} ${NEKTAR++_TP_LIBRARIES}
+          ${MPI_LIBRARIES} ${NESO_PARTICLES_LIBRARIES}
+          ${NESO_PARTICLES_LINK_FLAGS})
```

Comment:
Link flags shouldn't be included in the `target_link_library` command. Ideally they'd be passed transitively by linking against the NESO_PARTICLES target. Failing that, they should be set using [add_link_options](https://cmake.org/cmake/help/latest/command/add_link_options.html).

Reply:
Moved to `add_link_options` in the solver cmake and added a line in the NP issue [1] for a better cmake interface.

[1] https://github.com/ExCALIBUR-NEPTUNE/NESO-Particles/issues/35